### ISSUE TITLE
CA1420: "Requires runtime marshalling to be enabled," but it is enabled by default

### ISSUE THREAD

**Issue by ladenedge** [dotnet/roslyn-analyzers#6179]
### Analyzer

**Diagnostic ID**: [CA1420](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1420): `Delegates with managed types as parameters or the return type require runtime marshalling to be enabled in the assembly where the delegate is defined.`

 ⚠️ The above link is a 404, btw.

### Analyzer source

**SDK**: Built-in CA analyzers in .NET 7 RC1 SDK
**Version**: [SDK 7.0.100-rc.1.22431.12](https://dotnet.microsoft.com/en-us/download/dotnet/7.0)
**IDE**: Visual Studio v17.4.0 Preview 2.1

### Describe the bug

Using `GetDelegateForFunctionPointer` with managed parameters triggers this warning, but it does so even when runtime marshalling is _enabled_ for the assembly.

### Steps To Reproduce

```cs
[StructLayout(LayoutKind.Sequential)]
public class HeaderTable
{
	public bool ParseField(HeaderAccessorT accessor, ulong id, AptStrT value, IntPtr pool)
	{
		// CA1420 is triggered for the following line.
		var doParseField = Marshal.GetDelegateForFunctionPointer<ParseFieldDelegate>(parse_field);
		return doParseField(accessor, id, value, pool);
	}

	[return: MarshalAs(UnmanagedType.Bool)]
	delegate bool ParseFieldDelegate(HeaderAccessorT accessor, ulong id, AptStrT value, IntPtr pool);
}

```

### Expected behavior

When runtime marshalling is enabled, the warning should not appear.

### Actual behavior

The warning appears even when the assembly does not explicitly disable runtime marshalling.

## Additional context

To experiment, I added the following attribute to my code.

```cs
[assembly: DisableRuntimeMarshalling]
```

When I include that attribute, my program explodes quite promptly.  (Suggesting runtime marshalling is _enabled_ in the absence of that attribute.)

Let me know if I can provide any further information.  Thank you!

**Comment by jkoritzinsky** [comment]
This was fixed by https://github.com/dotnet/roslyn-analyzers/pull/6124, which will be in .NET 7 RC2.
